src/ResourceManager/Sql/Commands.Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs

```diff
@@ -0,0 +1,191 @@
+
+        protected const string SetByResourceIdParameterSet =
+            "Set a Managed Instance from an Azure resource id";
+
+        /// <summary>
+        /// Gets or sets the name of the Managed instance to use.
+        /// </summary>
+        [Parameter(
+            ParameterSetName = SetByNameAndResourceGroupParameterSet, 
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "SQL Database Managed instance name.")]
+        [Alias("Name")]
+        [ValidateNotNullOrEmpty]
+        public string ManagedInstanceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(ParameterSetName = SetByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// The new SQL administrator password for the Managed instance.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The new SQL administrator password for the Managed instance.")]
+        [ValidateNotNull]
+        public SecureString AdministratorPassword { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Managed instance License Type
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Determines which License Type of Sql Azure Managed Instance to use")]
+        [ValidateSet(Constants.LicenseTypeBasePrice, Constants.LicenseTypeLicenseIncluded, IgnoreCase = false)]
+        public string LicenseType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Storage Size in GB for Managed instance
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Determines how much Storage size to associate with Managed instance")]
+        public int? StorageSizeInGB { get; set; }
+
+        /// <summary>
+        /// Gets or sets the VCores for Managed instance
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Determines how much Vcore to associate with Managed instance")]
+        public int? Vcore { get; set; }
+
+        /// <summary>
+        /// The tags to associate with the Managed instance.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The tags to associate with the Managed instance.")]
+        [Alias("Tag")]
+        public Hashtable Tags { get; set; }
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "Generate and assign an Azure Active Directory Identity for this Managed instance for use with key management services like Azure KeyVault.")]
+        public SwitchParameter AssignIdentity { get; set; }
```

Comment:
This is what I expect :) New-MI is flipped

Reply:
Do you support UserAssigned Identities as well (I remember the answer being no, just checking)

Reply:
We do not support UserAssigned identities at the moment.